lib/spack/spack/repo.py

```diff
@@ -1271,16 +1269,22 @@ class UnknownPackageError(UnknownEntityError):
 
     def __init__(self, name, repo=None):
         msg = None
-        if repo:
-            msg = "Package '%s' not found in repository '%s'" % (name, repo)
-        else:
-            msg = "Package '%s' not found." % name
-
-        # special handling for specs that may have been intended as filenames
-        # prompt the user to ask whether they intended to write './<name>'
         long_msg = None
-        if name.endswith(".yaml"):
-            long_msg = "Did you mean to specify a filename with './%s'?" % name
+        if name:
+            if repo:
+                msg = \
+                    "Package '%s' not found in repository '%s'" % (name, repo)
```

Comment:
We usually try not to use backslash to go over multiple lines:
```suggestion
                msg = "Package '{0}' not found in repository '{1}'"
                msg = msg.format(name, repo)
```
OT: I don't think these kind of style rules not enforced by `flake8` are written anywhere and I look forward to the day we'll start using black.